models/users.js

```diff
@@ -259,6 +259,17 @@ const fetchUser = async ({ userId = null, username = null, githubUsername = null
         id = doc.id;
         userData = doc.data();
       });
+    } else if (discordId) {
+      const user = await userModel.where("discordId", "==", discordId).limit(1).get();
+      user.forEach((doc) => {
+        id = doc.id;
+        userData = doc.data();
+      });
+      const userStatusDocs = await userStatusModel.where("userId", "==", id).limit(1).get();
```

Comment:
here const userStatusDocs = await userStatusModel.where("userId", "==", id).limit(1).get();. The userStatusDocs variable is a list of documents. However, the userStatusDoc variable is only initialized with the first document in the list. This means that the data variable will only contain the data from the first document.